pkg/converter/serviceaccount_converter_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package converter_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/projectcalico/kube-controllers/pkg/converter"
+	api "github.com/projectcalico/libcalico-go/lib/apis/v3"
+
+	k8sapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var _ = Describe("ServiceAccount conversion tests", func() {
+
+	saConverter := converter.NewServiceAccountConverter()
+
+	It("should parse a Service Account to a Profile", func() {
+		sa := k8sapi.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "serviceaccount",
+				Labels: map[string]string{
+					"foo.org/bar": "baz",
+					"roger":       "rabbit",
+				},
+				Annotations: map[string]string{},
+			},
+		}
+
+		p, err := saConverter.Convert(&sa)
+		By("not generating a conversion error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		expectedName := "ksa.default.serviceaccount"
+		actualName := p.(api.Profile).Name
+		By("returning a Calico profile with the expected name", func() {
+			Expect(actualName).Should(Equal(expectedName))
+		})
+
+		// @casey: Do you think this check is useful?
```

Comment:
Seems fine to me, doesn't hurt to have this check :)

Reply:
cool! thanks